src/api/time/TimeContext.js

```diff
@@ -353,25 +365,298 @@ class TimeContext extends EventEmitter {
+    }
+
+    if (timeSystemOrKey === undefined) {
+      throw 'Please provide a time system';
+    }
+
+    let timeSystem;
+
+    if (typeof timeSystemOrKey === 'string') {
+      timeSystem = this.timeSystems.get(timeSystemOrKey);
+
+      if (timeSystem === undefined) {
+        throw `Unknown time system ${timeSystemOrKey}. Has it been registered with 'addTimeSystem'?`;
+      }
+    } else if (typeof timeSystemOrKey === 'object') {
+      timeSystem = timeSystemOrKey;
+
+      if (!this.timeSystems.has(timeSystem.key)) {
+        throw `Unknown time system ${timeSystemOrKey.key}. Has it been registered with 'addTimeSystem'?`;
+      }
+    } else {
+      throw 'Attempt to set invalid time system in Time API. Please provide a previously registered time system object or key';
+    }
+
+    this.system = this.#copy(timeSystem);
+    /**
+     * The time system used by the time
+     * conductor has changed. A change in Time System will always be
+     * followed by a bounds event specifying new query bounds.
+     *
+     * @event module:openmct.TimeAPI~timeSystem
+     * @property {TimeSystem} The value of the currently applied
+     * Time System
+     * */
+    this.emit(TIME_CONTEXT_EVENTS.timeSystemChanged, this.#copy(this.system));
+
+    if (bounds) {
+      this.setBounds(bounds);
+    }
+
+    return this.system;
+  }
+
+  /**
+   * Get the start and end time of the time conductor. Basic validation
+   * of bounds is performed.
+   * @returns {module:openmct.TimeAPI~TimeConductorBounds}
+   * @memberof module:openmct.TimeAPI#
+   * @method bounds
+   */
+  getBounds() {
+    //Return a copy to prevent direct mutation of time conductor bounds.
+    return this.#copy(this.boundsVal);
+  }
+
+  /**
+   * Set the start and end time of the time conductor. Basic validation
+   * of bounds is performed.
+   *
+   * @param {module:openmct.TimeAPI~TimeConductorBounds} newBounds
+   * @throws {Error} Validation error
+   * @fires module:openmct.TimeAPI~bounds
+   * @returns {module:openmct.TimeAPI~TimeConductorBounds}
+   * @memberof module:openmct.TimeAPI#
+   * @method bounds
+   */
+  setBounds(newBounds) {
+    const validationResult = this.validateBounds(newBounds);
+    if (validationResult.valid !== true) {
+      throw new Error(validationResult.message);
+    }
+
+    //Create a copy to avoid direct mutation of conductor bounds
+    this.boundsVal = this.#copy(newBounds);
+    /**
+     * The start time, end time, or both have been updated.
+     * @event bounds
+     * @memberof module:openmct.TimeAPI~
+     * @property {TimeConductorBounds} bounds The newly updated bounds
+     * @property {boolean} [tick] `true` if the bounds update was due to
+     * a "tick" event (i.e. was an automatic update), false otherwise.
+     */
+    this.emit(TIME_CONTEXT_EVENTS.boundsChanged, this.boundsVal, false);
+
+    //Return a copy to prevent direct mutation of time conductor bounds.
+    return this.#copy(this.boundsVal);
```

Comment:
Why does `setBounds()` return anything? Moreover, why does it take a _second_ copy of the bounds object?

Reply:
you know Andrew... that's a great question. I will remove it :)